Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -1531,40 +1545,48 @@ void TreeResolver::sortPositionOptionsIfNeeded(PositionOptions& options, const S
+    if (order != PositionTryOrder::Normal && options.optionStyles.size() > 2) {
+        // "For each entry in the position options list, apply that position option to the box, and find
+        // the specified inset-modified containing block size that results from those styles."
+        // https://drafts.csswg.org/css-anchor-position-1/#position-try-order-property
+        auto boxAxis = boxAxisForPositionTryOrder(order, options.optionStyles.at(0)->writingMode());
 
-    struct SortingOption {
-        std::unique_ptr<RenderStyle> style;
-        LayoutUnit containingBlockSize;
-    };
-    Vector<SortingOption> optionsForSorting;
-    optionsForSorting.reserveInitialCapacity(options.optionStyles.size());
+        struct SortingOption {
+            std::unique_ptr<RenderStyle> style;
+            LayoutUnit containingBlockSize;
+        };
+        Vector<SortingOption> optionsForSorting;
+        optionsForSorting.reserveInitialCapacity(options.optionStyles.size());
 
-    for (auto& optionStyle : options.optionStyles) {
-        auto constraints = PositionedLayoutConstraints { *box, *optionStyle, boxAxis };
-        constraints.computeInsets();
-        optionsForSorting.append({ WTFMove(optionStyle), constraints.insetModifiedContainingSize() });
-    }
+        for (size_t i = 1; i < options.optionStyles.size(); ++i) {
+            auto constraints = PositionedLayoutConstraints { *box, *options.optionStyles[i], boxAxis };
+            constraints.computeInsets();
+            optionsForSorting.append({ WTFMove(options.optionStyles[i]), constraints.insetModifiedContainingSize() });
+        }
 
-    // "Stably sort the position options list according to this size, with the largest coming first."
-    std::ranges::stable_sort(optionsForSorting, std::ranges::greater { }, &SortingOption::containingBlockSize);
+        // "Stably sort the position options list according to this size, with the largest coming first."
+        std::ranges::stable_sort(optionsForSorting, std::ranges::greater { }, &SortingOption::containingBlockSize);
 
-    for (size_t i = 0; i < optionsForSorting.size(); ++i)
-        options.optionStyles[i] = WTFMove(optionsForSorting[i].style);
+        for (size_t i = 0; i < optionsForSorting.size(); ++i)
+            options.optionStyles[i + 1] = WTFMove(optionsForSorting[i].style);
+    }
+
+    if (options.lastSuccessfulFallbackIndex) {
+        for (size_t i = 1; i < options.optionStyles.size(); ++i) {
+            if (options.optionStyles[i]->usedPositionTryFallbackIndex() == options.lastSuccessfulFallbackIndex) {
+                std::ranges::rotate(options.optionStyles, &options.optionStyles[i]);
+                break;
+            }
+        }
+    }
 }
 
-std::optional<ResolvedStyle> TreeResolver::tryChoosePositionOption(const Styleable& styleable, const RenderStyle* existingStyle)
+std::optional<ResolvedStyle> TreeResolver::tryChoosePositionOption(const Styleable& styleable)
```

Comment:
I like removing the existingStyle parameter here and passing things via PositionOptions.

Reply:
Thank you 🙏